operator/cache_memcached.go

```diff
@@ -161,10 +208,22 @@ func (m *MemcachedStore) ListKeys() []string {
 		return nil
 	}
 	keys := make([]string, 0)
-	m.keys.Range(func(key, value any) bool {
-		keys = append(keys, key.(string))
-		return true
-	})
+	item, err := m.client.Get(fmt.Sprintf(keysCacheKey, m.kind.Plural()))
```

Comment:
`ListKeys` is called every `CacheSync` interval (`CustomCacheInformer` defaults this to `0`/no cache sync, `KubernetesBasedInformer` actually defaults this to `30s` because of copy-pasted code way back, and we can probably change/remove this, as the syncs are eaten by the opinionated logic anyway). I can change this to use the in-memory map instead.